### PR TITLE
Add Temporal Terraform modules

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

All credits belong to @thehabbos007 

This adds two new modules to our terraform setup:
- Temporal: The temporal server and the related setup
- Temporal-Worker: The setup for the temporal workers

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1204274556967336/1204521571620722/f) _(internal)_

## 🔍 What does this change?

- Adjusts the migration and setup scripts to work in AWS
- Add module as mentioned above
- This does **not** add the modules to the production environment yet, this will be done in a follow-up

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing 

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] is internal and does not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] does not affect the execution graph 

## ⚠️ Known issues

- This currently uses a Docker-based Postgres instance instead of an RDS instance
- Authorization is not set up yet

## 🐾 Next steps

- Address the known issues
- Use module in production

@thehabbos007 added a few commits as an example setup. This PR removes these commits in order to merge the changes, but they will be used as guidance for further work.

### CODE REVIEW

File: infra/terraform/modules/temporal/task_definitions.tf [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+          "awslogs-group"         = local.log_group_name
+          "awslogs-stream-prefix" = local.setup_service_name
+          "awslogs-region"        = var.region
+        }
+      }
+    },
+  ]
+
+  shared_env_vars = [
+    { name = "DB", value = "postgres12" },
```
Comment (CiaranMn) [inline infra/terraform/modules/temporal/task_definitions.tf:158]: How is this used? If 12 is the version this doesn't match the version we are using in the local image defined in these files, or remotely (both 15)

Comment (CiaranMn) [reply infra/terraform/modules/temporal/task_definitions.tf]: Understood, thank you

File: infra/terraform/modules/temporal/main.tf [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    description = "Allow connections to Postgres within the VPC"
+    cidr_blocks = [var.vpc.cidr_block]
+  }
+
+  ingress {
+    from_port   = 7233
+    to_port     = 7233
+    protocol    = "tcp"
+    description = "Allow connections to Temporal within the VPC (load balancer)"
+    cidr_blocks = ["0.0.0.0/0"]
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply infra/terraform/modules/temporal/main.tf:178]: I don't know yet if we want to make it private. Are you fine with these changes?
```suggestion
    description = "Allow connections to Temporal from anywhere (rely on authentication to restrict access)"
    # TODO: Consider changing this to `var.vpc.cidr_block`
    cidr_blocks = ["0.0.0.0/0"]
```

File: infra/terraform/hash/hash_application/main.tf [hunk 3]
```diff
@@ -26,25 +26,12 @@ locals {
...
   }
 }
 
-data "aws_security_group" "vpce" {
-  filter {
-    name   = "group-name"
-    values = ["${var.prefix}-sgvpce"]
-  }
-}
-
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply infra/terraform/hash/hash_application/main.tf:47]: They were used as part of [`48295f7` (#2713)](https://github.com/hashintel/hash/pull/2713/commits/48295f7a22596d0c6570e01ba61f58532a0a2ca0). We currently don't use them.

File: infra/terraform/modules/temporal/main.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (CiaranMn) [reply infra/terraform/modules/temporal/main.tf:178]: Yep, looks good

File: infra/terraform/hash/hash_application/main.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (CiaranMn) [reply infra/terraform/hash/hash_application/main.tf:47]: Got it, thank you

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**